### PR TITLE
[Testing] Add PhanTypeVoidAssignment to static analysis

### PR DESCRIPTION
**Pull request by johnsaigle**
This rule prevents the variables being assigned to functions that are declared to return void, which makes sense.

### CODE REVIEW

File: php/libraries/NDB_Form.class.inc [hunk 1]
```diff
@@ -54,7 +54,7 @@ class NDB_Form extends NDB_Page
      * @param string $page       Identifies the page to show
      * @param string $identifier Variables passed to form
      *
-     * @return void
+     * @return object
```
Comment (driusan) [inline php/libraries/NDB_Form.class.inc]:
```suggestion
     * @return \NDB_Form
```

I think the factory for an NDB_Form should always return some type of NDB_Form, not a generic object.

File: modules/statistics/php/statistics_mri_site.class.inc [hunk 2]
```diff
@@ -73,9 +73,9 @@ class Statistics_Mri_Site extends Statistics_Site
      * @param string $projectID the value of projectID
      * @param string $issue     the value of issue
      *
-     * @return void
+     * @return ?string
      */
-    function _completeCount($centerID, $projectID, $issue)
+    function _completeCount($centerID, $projectID, $issue): ?string
```
Comment (PapillonMcGill) [inline modules/statistics/php/statistics_mri_site.class.inc]: line 92   _getResults()  add **:array**

Comment (PapillonMcGill) [reply modules/statistics/php/statistics_mri_site.class.inc]: And stressed that same point in another PR :-)

**Comment by driusan** [comment]
#4283 is merged.

File: modules/statistics/php/statistics_mri_site.class.inc [hunk 3]
```diff
@@ -209,6 +209,8 @@ class Statistics_Mri_Site extends Statistics_Site
 
         $data = array();
         foreach ($this->issues as $issue) {
+            // FIXME The function _completeCount() seems ot be unfinished and
```
[the remark this answers is not in the document]
Comment (driusan) [reply modules/statistics/php/statistics_mri_site.class.inc]: @johnsaigle

**Pull request merged**